### DISCUSSION TITLE
keycloak Integrates permissions

### DISCUSSION THREAD

**Discussion by holyld1987** [gravitational/teleport#17947] [Q&A] [answered]
Hello, our company uses the keycloak permission system, and now our company wants to integrate teleport into our system. In order to unify the permission system, I will not remove the open source teleport branch, which does not support access to keycloak. If so, what should I do? Thank you for the

**Comment by webvictim** [comment, marked as answer]
Keycloak offers an OIDC interface which Teleport Enterprise can integrate with.

If you'd like to get more information on Teleport Enterprise, check out this page here: https://goteleport.com/pricing/

**Comment by holyld1987** [comment]
Hi,thanks for the reply! I have another question to ask you. My company deployed the open source precompiled teleport, and now I compile with the teleport source code. The company has added a lot of data on teleport, and now they want to use the system compiled with my source code. I would like to ask how I can transfer the data from the company's pre-compiled system to my current system, thank you!

**Reply by webvictim** [reply]
If you just replace the precompiled binary in `/usr/local/bin/teleport` with your compiled one from `build/teleport` and then restart Teleport, that will do it.

**Reply by holyld1987** [reply]
thanks for your reply! i have another question ,Can open source projects set up anonymous access?thanks

**Reply by webvictim** [reply]
I'm not 100% sure what you mean by anonymous access, but to use Teleport you must always have a user provisioned by the cluster administrator or a single sign-on provider - there is no ability to bypass logging in.
